### PR TITLE
Account for missing fields in Rollout HealthStatus

### PR DESCRIPTION
**Pull request by dthomson25**
Address https://github.com/argoproj/argo-cd/issues/1697

### CODE REVIEW

**Review by alexec — CHANGES_REQUESTED** [review]
1. You're several weeks behind master. `git merge master` please.
2. IMHO, this needs tests.

File: resource_customizations/argoproj.io/Rollout/health_test.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 tests:
+- healthStatus:
```
Comment (alexec) [inline resource_customizations/argoproj.io/Rollout/health_test.yaml]: Am I an idiot - is this the test?

**Comment by jessesuen** [comment]
Can this be simplified if we drop v0.2 rollout support?

File: resource_customizations/argoproj.io/Rollout/health_test.yaml (hunk 1 above)
Comment (dthomson25) [reply resource_customizations/argoproj.io/Rollout/health_test.yaml]: Yes this is the test. We have that documented here: https://argoproj.github.io/argo-cd/operator-manual/health/#way-2-contribute-a-custom-health-check but I can see how that can be confusing. Honestly, I think we may want to rethink this testing framework. The current tests are valuable but I found it difficult to maintain.

**Comment by dthomson25** [comment]
@jessesuen it would simplify the overall script by a few lines but that would be unrelated to the change i made. A lot of the complexity comes from trying to provide a message of the state of the rollout in the health status.

File: resource_customizations/argoproj.io/Rollout/health_test.yaml (hunk 1 above)
Comment (alexec) [reply resource_customizations/argoproj.io/Rollout/health_test.yaml]: cool - you still need to sync with master

**Review by jessesuen — APPROVED** [review]

**Comment by alexec** [comment]
You've got Jesse's approval - so you can dismis mine and merge.

**Review by alexec — APPROVED** [review]

**Pull request merged**